search-sentiments-map/public/scripts/modal-script.js

```diff
@@ -105,50 +106,71 @@ async function displayTopResultsForCurrentTrend(countryCode) {
-                    .includes(checkWordInArray)) {
-                  resultElement.innerHTML += '<span style=\'color: ' + 
-                      NEGATIVE_COLOR + ';\'>' + word + ' </span>';
-                } else {
-                  resultElement.innerHTML += word + ' ';
-                }
-              });
-              resultElement.innerHTML += '<br><i>Sentiment Score: ' + 
-                  results[i].score.toFixed(1) + '</i><br>';
-            } else {
-              resultElement.innerHTML += results[i].snippet + '<br>' + 
-                  '<i>Sentiment Score: ' + results[i].score.toFixed(1) + 
-                  '</i><br>';
+
+            // Check that positive and negative words are detected by the Node.js 
+            // sentiment API.
+            if (sentimentWordsResult != null) {
+              let positiveWords = sentimentWordsResult.positive;
+              let negativeWords = sentimentWordsResult.negative;
+              if (results[i].score > 0 && positiveWords.length !== 0) {
+                highlightWords(snippet, new Set(positiveWords), true, 
+                    resultElement);
+              } else if (results[i].score < 0 && negativeWords.length !== 0) {
+                highlightWords(snippet, new Set(negativeWords), false, 
+                    resultElement);
+              } else {
+                resultElement.innerHTML += snippet;
+              }
             }
+            resultElement.innerHTML += '<br><i>Sentiment Score: ' + 
+                  results[i].score.toFixed(1) + '</i><br>';
           });
     }
     resultElement.innerHTML += '<i>Last updated on ' + date + '</i>';
   }
 }
 
+/** 
+ * Highlights the positive words green if the sign of the sentiment score 
+ * is positive and negative words red if the sign is negative.
+ * @param {string} snippet Snippet to display in the search results tab.
+ * @param {Set} wordsToHighlight Words to highlight a different color.
+ * @param {boolean} scoreIsPositive Boolean for whether or not the sentiment 
+ *     score is positive.
+ * @param {Object} resultElement Tab element to update with the search result 
+ *     snippet.
+ */
+function highlightWords(
+    snippet, wordsToHighlight, scoreIsPositive, resultElement) {
+  let startDisplayingSnippetIndex = 0;
+  wordsToHighlight.forEach((word) => {
+    let wordIndex = snippet.toLowerCase().indexOf(word);
+    if (wordIndex !== -1){
+      let positiveHighlight = '<span style="color: ' +
+            POSITIVE_COLOR + ';">' + snippet.substring(wordIndex, 
+            wordIndex + word.length) + '</span>';
+      let negativeHighlight = '<span style="color: ' +
+            NEGATIVE_COLOR + ';">' + snippet.substring(wordIndex, 
+            wordIndex + word.length) + '</span>';
+      snippet = scoreIsPositive ? snippet.replace(new RegExp(
+          '\\b'+ word + '\\b', 'gi'), positiveHighlight) : 
+          snippet.replace(new RegExp('\\b' + word + '\\b', 'gi'), 
+          negativeHighlight);
```

Comment:
I think it should be something like this

@ChenyuHeidiZhang could you double check too?

```suggestion
            POSITIVE_COLOR + ';">' +
            snippet.substring(wordIndex, wordIndex + word.length) +
            '</span>';
      let negativeHighlight = '<span style="color: ' +
            NEGATIVE_COLOR + ';">' +
            snippet.substring(wordIndex,  wordIndex + word.length) +
            '</span>';
      snippet = scoreIsPositive ?
          snippet.replace(new RegExp( '\\b'+ word + '\\b', 'gi'), positiveHighlight) : 
          snippet.replace(new RegExp('\\b' + word + '\\b', 'gi'), negativeHighlight);
```

Reply:
I just checked, and you're correct, I knew we had to break highest syntactically, but I wasn't sure what to do then because the line wasn't broken at those syntatic points. I just looked at the example again, and it's similar to what you do here. Thanks!
https://google.github.io/styleguide/jsguide.html#formatting-where-to-break